### PR TITLE
edl: add udev rules

### PR DESCRIPTION
**Pull request by ajaxbits**
## Description of changes
Adds udev rules to the package, for consumption by NixOS's `services.udev.packages`. This is part of the [installation instructions](https://github.com/bkerler/edl?tab=readme-ov-file#linux-debianubuntumintetc) for `edl`.

Also removes some unused `rec`s, comments, and reorders attributes according to `nixpkgs-hammering`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by h7x4 — APPROVED** [review]
This seems reasonable, based on the diff

**Review by lorenz — APPROVED** [review]
LGTM

**Comment by lorenz** [comment]
Actually on second thought I'm not sure that we should include 50-android.rules as that is canonically maintained in the android-udev-rules package.

**Comment by ajaxbits** [comment]
> Actually on second thought I'm not sure that we should include 50-android.rules as that is canonically maintained in the android-udev-rules package.

Thanks for pointing that out. I'll make that change when I have time at my desk next.

File: pkgs/development/embedded/edl/default.nix [hunk 1]
```diff
@@ -28,11 +20,27 @@ python3Packages.buildPythonPackage rec {
...
+  # edl has a spurious dependency on "usb" which has nothing to do with the
+  # project and was probably added by accident trying to add pyusb
+  postPatch = ''
+    sed -i '/'usb'/d' setup.py
+  '';
+
+  postInstall = ''
+    mkdir -p $out/etc/udev/rules.d
+    cp $src/Drivers/50-android.rules $out/etc/udev/rules.d/50-android.rules
+    cp $src/Drivers/51-edl.rules $out/etc/udev/rules.d/51-edl.rules
```
Comment (Mic92) [inline pkgs/development/embedded/edl/default.nix:41]:
```suggestion
    mkdir -p $out/etc/udev/rules.d
    cp $src/Drivers/51-edl.rules $out/etc/udev/rules.d/51-edl.rules
```

**Review by lorenz — APPROVED** [review]

**Pull request merged**